test/src/test/java/jenkins/slaves/OldRemotingAgentTest.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.slaves;
+
+import hudson.EnvVars;
+import hudson.model.Computer;
+import hudson.model.FreeStyleProject;
+import hudson.model.Label;
+import hudson.model.Slave;
+import hudson.model.labels.LabelAtom;
+import hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor;
+import hudson.node_monitors.AbstractNodeMonitorDescriptor;
+import hudson.node_monitors.NodeMonitor;
+import hudson.slaves.ComputerLauncher;
+import hudson.tasks.BatchFile;
+import hudson.tasks.Shell;
+import org.codehaus.plexus.util.FileUtils;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.SimpleCommandLauncher;
+
+import javax.annotation.CheckForNull;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.URISyntaxException;
+import java.util.Collection;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for old Remoting agent versions
+ */
+public class OldRemotingAgentTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRuleWithOldAgent();
+
+    @Rule
+    public TemporaryFolder tmpDir = new TemporaryFolder();
+
+    private File agentJar;
+
+    @Before
+    public void extractAgent() throws Exception {
+        agentJar = new File(tmpDir.getRoot(), "old-agent.jar");
+        FileUtils.copyURLToFile(getClass().getResource("/old-remoting/remoting-minimal-supported.jar"), agentJar);
+    }
+
+    @Test
+    @Issue("JENKINS-48761")
```

Comment:
This test fails on Jenkins 2.99/98. `ChannelClosedException` API compatibility is not tested though